imagescan/delta.go

```diff
@@ -173,13 +180,12 @@ func (d *deltaState) upsertImages(pod *corev1.Pod) {
 
 		nodeName := pod.Spec.NodeName
 		arch := d.getPodArch(pod)
-		key := d.getImageKey(cs.ImageID, arch)
+		key := newImageKey(cs.ImageID, arch, cont.Image)
 		img, found := d.images[key]
 		if !found {
-			img = newImage(key, cs.ImageID, arch)
+			img = newImage(cont.Image, cs.ImageID, arch)
```

Comment:
It looks strange to create key twice here and inside newImage. I think the problem that newImage constructor takes args. 

Since we still need to fill other image fields, this will simplify it and make more clear that newImage just returns placeholder object with initialized maps and backoff. Then we can drop newImageKey func to.
```go
key := cs.ImageID + cs.Image + arch
img, found := d.images[key]
if !found {
	img = newImage()
	img.name = cont.Name
	img.key = key
	img.architecture = arch
}
```

```